Lib/json/tool.py

```diff
@@ -68,12 +70,42 @@ def main():
             outfile = open(options.outfile, 'w', encoding='utf-8')
         with outfile:
             for obj in objs:
-                json.dump(obj, outfile, **dump_args)
+                if can_colorize(file=outfile):
+                    json_str = json.dumps(obj, **dump_args)
+                    outfile.write(colorize_json(json_str))
+                else:
+                    json.dump(obj, outfile, **dump_args)
                 outfile.write('\n')
     except ValueError as e:
         raise SystemExit(e)
 
 
+color_pattern = re.compile(r'''
```

Comment:
Let's say that this pattern assumes that the matched string is a valid JSON, hence the more flexible (and readable) regex.

Reply:
Added in [b5157af](https://github.com/python/cpython/pull/132126/commits/b5157af6eb8822d3c9daa6d13c2670d6af12dcc1)